From 1de4f37c8da084027e7393cdb46824c530ceb0db Mon Sep 17 00:00:00 2001
From: Larry Gritz <lg@larrygritz.com>
Date: Tue, 10 Dec 2019 09:25:32 -0800
Subject: [PATCH] Exif read: guard better against out of range offests (#2429)

The dataptr() function returns nullptr if the offset is out of proper
range. But the array_to_spec function calling dataptr did not check for
a nullptr return, ended up segfaulting.

This can happen with JPEG files that have certain malformed (we think)
sections of their camera maker notes.
---
 src/libOpenImageIO/exif-canon.cpp | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/libOpenImageIO/exif-canon.cpp b/src/libOpenImageIO/exif-canon.cpp
index f098c19b4..335540540 100644
--- a/src/libOpenImageIO/exif-canon.cpp
+++ b/src/libOpenImageIO/exif-canon.cpp
@@ -634,6 +634,8 @@ array_to_spec (ImageSpec& spec,                 // spec to put attribs into
         ASSERT(0 && "unsupported type");
     }
     const T *s = (const T *) pvt::dataptr (dir, buf, offset_adjustment);
+    if (!s)
+        return;
     for (auto&& attr : indices) {
         if (attr.value < int(dir.tdir_count)) {
             T ival = int (s[attr.value]);
